wallet-new/integration/Functions.hs

```diff
@@ -326,6 +336,32 @@ runAction wc ws GetTransaction  = do
     -- Modify wallet state accordingly.
     pure $ ws
         & actionsNum +~ 1
+runAction wc ws ImportPoorWallet = do
+    poorSecret <- pickRandomElement =<< liftIO unsafePoorSecrets
+    withSystemTempFile "importWalletKey.sk" $ \tempFp _ -> do
```

Comment:
This seems too complicated. What we need is to import a key to the wallet. 
If we are generating keys every time we run this function this might be too expensive.
We can generate the import wallet before this test, and that would be simpler.
I mean, we can have a single key to import and import it once before the test and that would be good enough.

Reply:
Yes - my understanding what @parsonsmatt said here https://iohk.myjetbrains.com/youtrack/issue/CSL-2371#comment=93-19258 is exactly aligned with what you are proposing.

I will ditch this code and expect `FilePath` to keys on file system will be given (and known upfront) when we will have this service ready. 

My understanding is that we don't have mechanism so far to generate keys on system startup, ie on this step:
> Test suite boots up the server, with initial configuration options

so maybe I can generate keys as a part of `wallet-new/integration/Main.hs` when test suite is starting up. What do you think? 

EDIT:

On a second thought I believe what @parsonsmatt whanted to say there is to import wallet using V0 endpoint there - and then this wallet can be used for other integration tests. But I am not sure what he exactly meant (maybe you can help me clear the picture).

How I think we should implement it:
 * generate keys as a part of starting up integration test suite - in `wallet-new/integration/Main.hs`
 * when V1.importKey is implemented - add integration test that imports keys using `FilePath` of previously generated keys

Otherwise I am not sure why we need import key in integration tests at all atm. Can you help me explain the motivation @ksaric ?

Reply:
@akegalj The motivation is that we need money when we are running this integration test so we can send transactions around.

Reply:
Ok, in that case I will just do what I have described:
 * generate keys as a part of starting up integration test suite - in wallet-new/integration/Main.hs
 * when V1.importKey is implemented - add integration test that imports keys using FilePath of previously generated keys

I have wrongly assumed the motivation is to test V0.importKey that will eventually be in place. Ok - will fix it now

Reply:
Personally, I don't see anything wrong with this test (it will lead to more random states), but the key generation should be extracted outside it in any case.

Reply:
> generate keys as a part of starting up integration test suite - in wallet-new/integration/Main.hs

:+1: 

> when V1.importKey is implemented - add integration test that imports keys using FilePath of previously generated keys

Sounds good!

> I have wrongly assumed the motivation is to test V0.importKey that will eventually be in place. Ok - will fix it now

To further clarify: the wallet must be initialized so that it has money we can spend with. *How* that happens should be kept out of the tests themselves -- as you've noted, we can do that in `main` before we actually start running the test suite. We can do an `http-client` request to the `V0` part of the API; or we can do something else (maybe there's a command line option to start the wallet with a key? I'm unfamiliar).

At some point, we'll have implemented the `v1/import-wallet` endpoint, at which point we can write a full integration test like this.